src/net.cpp

```diff
@@ -2753,8 +2753,7 @@ CNode::~CNode()
 {
     CloseSocket(hSocket);
 
-    if (pfilter)
-        delete pfilter;
+    delete pfilter;
```

Comment:
BTW as this one is a) in a destructor b) at the end of the destructor, why aren't we using a smart pointer w/ RAII?